repose-aggregator/components/filters/openstack-identity-v3/src/test/scala/org/openrepose/filters/openstackidentityv3/OpenStackIdentityV3HandlerTest.scala

```diff
@@ -291,35 +291,56 @@ class OpenStackIdentityV3HandlerTest extends FunSpec with BeforeAndAfter with Ma
     }
 
     it("should provide all the projects when the flag says to write all") {
-      val randomness = java.util.UUID.randomUUID.toString
-      identityV3Handler invokePrivate writeProjectHeader(randomness, roles, true, filterDirector)
-      verify(headerManager).appendHeader(org.mockito.Matchers.eq("X-Project-ID"), org.mockito.Matchers.eq("12345"), org.mockito.Matchers.eq("67890"), org.mockito.Matchers.eq(randomness))
+      val defaultPid = java.util.UUID.randomUUID.toString
+      val uriPid = java.util.UUID.randomUUID.toString
+      identityV3Handler invokePrivate writeProjectHeader(Some(defaultPid), roles, Some(uriPid), true, false, headerManager)
+      verify(headerManager).appendHeader(org.mockito.Matchers.eq("X-Project-ID"), org.mockito.Matchers.eq(defaultPid))
+      verify(headerManager).appendHeader(org.mockito.Matchers.eq("X-Project-ID"), org.mockito.Matchers.eq("12345"))
+      verify(headerManager).appendHeader(org.mockito.Matchers.eq("X-Project-ID"), org.mockito.Matchers.eq("67890"))
     }
 
     it("should handle when roles are present but project ids are missing") {
-      val randomness = java.util.UUID.randomUUID.toString
+      val defaultPid = java.util.UUID.randomUUID.toString
+      val uriPid = java.util.UUID.randomUUID.toString
       val roleList = List(Role(null, null, None, None, None), Role(null, null, None, None, None))
-      identityV3Handler invokePrivate writeProjectHeader(randomness, roleList, true, filterDirector)
-      verify(headerManager).appendHeader(org.mockito.Matchers.eq("X-Project-ID"), org.mockito.Matchers.eq(randomness))
+      identityV3Handler invokePrivate writeProjectHeader(Some(defaultPid), roleList, Some(uriPid), true, false, headerManager)
+      verify(headerManager).appendHeader(org.mockito.Matchers.eq("X-Project-ID"), org.mockito.Matchers.eq(defaultPid))
     }
 
     it("should handle when project ids are missing and not missing") {
-      val randomness = java.util.UUID.randomUUID.toString
+      val defaultPid = java.util.UUID.randomUUID.toString
+      val uriPid = java.util.UUID.randomUUID.toString
       val roleList = List(Role(null, null, None, None, None), Role(null, null, Option("foo"), None, None), Role(null, null, None, None, None))
-      identityV3Handler invokePrivate writeProjectHeader(randomness, roleList, true, filterDirector)
-      verify(headerManager).appendHeader(org.mockito.Matchers.eq("X-Project-ID"), org.mockito.Matchers.eq("foo"), org.mockito.Matchers.eq(randomness))
+      identityV3Handler invokePrivate writeProjectHeader(Some(defaultPid), roleList, Some(uriPid), true, false, headerManager)
+      verify(headerManager).appendHeader(org.mockito.Matchers.eq("X-Project-ID"), org.mockito.Matchers.eq(defaultPid))
+      verify(headerManager).appendHeader(org.mockito.Matchers.eq("X-Project-ID"), org.mockito.Matchers.eq("foo"))
+    }
+
+    it("should add qualities when flagged and not writing all") {
```

Comment:
So currently it won't break anything, but that doesn't mean it'll always be the case, and i've seen many customers have random crap turned on in there config because they are having problems getting something to do what they expect and just start doing random crap without understanding the consequences. If we can safegaurd against i'm all for that. Again a decision i don't agree with while i'm not present.